analyzers/tests/SonarAnalyzer.UnitTest/Rules/CalculationsShouldNotOverflowSyntaxKindWalkerTest.cs

```diff
@@ -0,0 +1,241 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using CalculationsShouldNotOverflow = SonarAnalyzer.SymbolicExecution.Roslyn.RuleChecks.CSharp.CalculationsShouldNotOverflow;
+
+namespace SonarAnalyzer.UnitTest.Rules;
+
+[TestClass]
+public class CalculationsShouldNotOverflowSyntaxKindWalkerTest
+{
+    [DataTestMethod]
+    [DataRow("_ = i + i;")]
+    [DataRow("_ = i - i;")]
+    [DataRow("_ = i * i;")]
+    [DataRow("_ = ++i;")]
+    [DataRow("_ = --i;")]
+    [DataRow("_ = i++;")]
+    [DataRow("_ = i--;")]
+    [DataRow("i += 1;")]
+    [DataRow("i -= 1;")]
+    [DataRow("i *= 1;")]
+    [DataRow("_ = (i + i, 0);")]
+    [DataRow("""
+        _ = i + i;
+        _ = 1 / i;
+        """)]
+    [DataRow("""
+        _ = ++i;
+        _ = +i;
+        """)]
+    [DataRow("""
+        _ = i++;
+        _ = i!;
+        """)]
+    [DataRow("""
+        i += 1;
+        i /= 1;
+        """)]
+    public void HasOverflowExpressions_True(string statement)
+    {
+        var tree = TestHelper.CompileCS(WrapInMethod($"""
+            var i = 0;
+            {statement}
+            """)).Tree;
+        var sut = new CalculationsShouldNotOverflow.SyntaxKindWalker();
+        sut.SafeVisit(tree.GetRoot());
+        sut.HasOverflow.Should().Be(true);
+        sut.IsUnchecked.Should().Be(false);
+    }
+
+    [DataTestMethod]
+    [DataRow("")]
+    [DataRow("_ = i / 1;")]
+    [DataRow("_ = i % 1;")]
+    [DataRow("_ = i == 1;")]
+    [DataRow("_ = i != 1;")]
+    [DataRow("_ = i > 1;")]
+    [DataRow("_ = i >= 1;")]
+    [DataRow("_ = i < 1;")]
+    [DataRow("_ = i <= 1;")]
+    [DataRow("_ = i >>> 1;")]
+    [DataRow("_ = i << 1;")]
+    [DataRow("_ = i | 1;")]
+    [DataRow("_ = i & 1;")]
+    [DataRow("_ = b && b;")]
+    [DataRow("_ = b || b;")]
+    [DataRow("_ = b ^ b;")]
+    [DataRow("_ = null ?? new object();")]
+    [DataRow("_ = i is int;")]
+    [DataRow("_ = i as int?;")]
+    [DataRow("i /= 1;")]
+    [DataRow("i %= 1;")]
+    public void HasOverflowExpressions_False(string statement)
+    {
+        var tree = TestHelper.CompileCS(WrapInMethod($"""
+            var i = 0; var b = true;
+            {statement}
+            """)).Tree;
+        var sut = new CalculationsShouldNotOverflow.SyntaxKindWalker();
+        sut.SafeVisit(tree.GetRoot());
+        sut.HasOverflow.Should().Be(false);
```

Comment:
There're dedicated methods for true and false. Same below.
```suggestion
        sut.HasOverflow.Should().BeFalse();
```